### PR TITLE
EXC-GI_Pset_ImprovedGroundTypeDewatered plus enum IfcImprovedGroundTypeEnum.DEWATERED

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_ImprovedGroundTypeDewatered

## Description:

Properties describing improved ground of type dewatered.

| Property                | Datatype                     | Description                                                  |
| ----------------------- | ---------------------------- | ------------------------------------------------------------ |
| DewateringMethod        | PEnum_DewateringMethod       | Active dewatering of the ground through pumping              |
| UserDefinedDewateringMethod  | IfcLabel                     | Encompasses custom or specialized dewatering techniques beyond standard categories. Used when DewateringMethod = USERDEFINED. |
| StaticSuctionLift       | IfcPositiveLengthMeasure             | Measurement of the vertical distance between the water table and the suction point. |
| StaticDischargeHead     | IfcPositiveLengthMeasure             | Measurement of the vertical distance from the discharge point to a reference elevation. |
| VolumetricFlowRate      | IfcVolumetricFlowRateMeasure | The volumetric flow rate of the fluid. **Existing property** |
| BoreholeCount       | IfcInteger                   | Number of boreholes.                     |
| AverageBoreholeDepth | IfcPositiveLengthMeasure     | Average depth of boreholes used for dewatering. |
| BoreholeSpacing      | IfcPositiveLengthMeasure     | Spacing of boreholes used for dewatering. |

## Enumeration: PEnum_DewateringMethod

Enumeration of dewatering methods.

| Constant              | Description                        |
| --------------------- | ---------------------------------- |
| SUMPPUMP              | Utilization of pumps to remove water collected in a sump or low point.  |
| WELLPOINT             | System of closely spaced wells connected to a common header.  |
| DEEPWELL              | Utilization of drilled wells to lower the water table in deeper soil layers.  |
| VACUUMWELLPOINTS      | Wellpoint system augmented by vacuum technology for increased dewatering efficiency.  |
| EDUCTORWELLS          | Usage of eductor pumps to induce flow and lift water for dewatering.  |
| USERDEFINED | **Existing constant** |

### CODE REVIEW

**Comment by larswik** [comment]
Check enum descriptions. Ping @JontyDunn and @hessf-ilf

File: IFC4x3/Properties/a/AverageDepthOfBoreholes_3NM6PipTTFyvqSdj8magOF/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AverageDepthOfBoreholes_3NM6PipTTFyvqSdj8magOF" Name="AverageDepthOfBoreholes" UniqueId="d758666c-cdd7-4ff3-9d1c-9ed23092a60f" PrimaryDataType="IfcPositiveLengthMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/a/AverageDepthOfBoreholes_3NM6PipTTFyvqSdj8magOF/DocProperty.xml]: Propose renaming to "AverageBoreholDepth".

File: IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfBoreholes_1DBeislK1CbfTPnPGztE29" Name="NumberOfBoreholes" UniqueId="4d2e8b36-bd40-4c96-9759-c5943ddce089" PrimaryDataType="IfcInteger" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/DocProperty.xml]: This is definitely already defined. Same proposal to rename to "BoreholeCount" and having the same type. Either `IfcInteger` or `IfcCountMeasure`.

File: IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/Documentation.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Number of boreholes used for dewatering.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/Documentation.md]: Too specific for a generic property name.

File: IFC4x3/Properties/s/SpacingOfBoreholes_3zHZ8zixv0VPAwab6YkHPN/DocProperty.xml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SpacingOfBoreholes_3zHZ8zixv0VPAwab6YkHPN" Name="SpacingOfBoreholes" UniqueId="fd46323d-b3be-407d-92ba-9251a2b91657" PrimaryDataType="IfcPositiveLengthMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SpacingOfBoreholes_3zHZ8zixv0VPAwab6YkHPN/DocProperty.xml]: BoreholeSpacing? Or maybe RadialSpacing as in #677?

File: IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Static discharge head.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md]: Tough to get something useful from this description.

File: IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Static suction lift.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md]: Description. 😞

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcImprovedGroundTypeEnum/DocEnumeration.xml [hunk 7]
```diff
@@ -10,6 +10,7 @@
 		<DocConstant xsi:nil="true" href="DEEPMIXED_14tVPAsoP2txdXAOYfnl1T" />
 		<DocConstant xsi:nil="true" href="LATERALLYDRAINED_3nfik2BfzA7fXPiEi_Zj8M" />
 		<DocConstant xsi:nil="true" href="FROZEN_0W7s0LZvb3mw4M6kTGSKdj" />
+		<DocConstant xsi:nil="true" href="DEWATERED_1BDq2sVVj5w9uDrfSeaMBh" />
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcImprovedGroundTypeEnum/DocEnumeration.xml:14]: Oh? A schema change even? Alright. 😄

File: IFC4x3/PropertyEnumerations/d/PEnum_DewateringMethod.xml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_DewateringMethod" Name="PEnum_DewateringMethod" UniqueId="18f441bb-53f6-43dd-ac9f-1f31a328bd05">
+	<Documentation>Enumeration of dewatering methods.</Documentation>
+	<Constants>
+		<DocPropertyConstant xsi:nil="true" href="SUMPPUMP_1DUgjYAoH4Gh7jVsG9H47C" />
+		<DocPropertyConstant xsi:nil="true" href="WELLPOINT_3HsrVdJYjAsOoHKTQjQhpE" />
+		<DocPropertyConstant xsi:nil="true" href="DEEPWELL_2PwZhTskj2V8OfGWvFYuKw" />
+		<DocPropertyConstant xsi:nil="true" href="VACUUMWELLPOINTS_1Ut4uPtD93cPyjtvmBw5Sq" />
+		<DocPropertyConstant xsi:nil="true" href="EDUCTORWELLS_1LL9tl1_j8B8nYd0uYcCEf" />
+		<DocPropertyConstant xsi:nil="true" href="OTHERDEWATERINGMETHOD_0XkzIJfrf01A4wQNSilK0U" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/d/PEnum_DewateringMethod.xml]: Propose https://github.com/bSI-InfraRoom/IFC-Specification/pull/564#pullrequestreview-1748790820

**Review by SergejMuhic — COMMENTED** [review]
- Constant descriptions

**Comment by hessf-ilf** [comment]
Agree with Sergej's comments.

> Should this be IfcPositiveLengthMeasure?

Yes for both accounts.

> Existing property which is defined as a bounded value (min-max). Is this ok?

Sounds good to me. As long as it's possible to set the min and the max as the same number.

> Assuming positive values

Again, yes. Depth and spacing may not be negative,

File: IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md]: Ping @JontyDunn and @hessf-ilf

File: IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md (hunk 6 above)
Comment (larswik) [reply IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md]: Ping @JontyDunn and @hessf-ilf

**Comment by hessf-ilf** [comment]
> Check enum descriptions. Ping @JontyDunn and @hessf-ilf

One possible proposition:

1. **SUMPPUMP:**
   -  Utilization of pumps to remove water collected in a sump or low point.

2. **WELLPOINT:**
   - System of closely spaced wells connected to a common header.

3. **DEEPWELL:**
   - Utilization of drilled wells to lower the water table in deeper soil layers.

4. **VACUUMWELLPOINTS:**
   - Wellpoint system augmented by vacuum technology for increased dewatering efficiency.

5. **EDUCTORWELLS:**
   - Usage of eductor pumps to induce flow and lift water for dewatering.

6. **OTHERDEWATERINGMETHOD:**
   - Encompasses custom or specialized dewatering techniques beyond standard categories.

File: IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md (hunk 5 above)
Comment (hessf-ilf) [reply IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md]: "Measurement of the vertical distance from the discharge point to a reference elevation." ?

File: IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md (hunk 6 above)
Comment (hessf-ilf) [reply IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md]: "Measurement of the vertical distance between the water table and the suction point." ?

File: IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/DocProperty.xml]: It is not defined in the current spec as far as I can see. Renaming to BoreholeCount.

File: IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/Documentation.md (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/Documentation.md]: A proposal would have been nice ;-) What about just shortening it to "Number of boreholes" and assume that the Pset provides the context?

File: IFC4x3/Properties/a/AverageDepthOfBoreholes_3NM6PipTTFyvqSdj8magOF/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/a/AverageDepthOfBoreholes_3NM6PipTTFyvqSdj8magOF/DocProperty.xml]: Done in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

File: IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/DocProperty.xml]: Done in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

File: IFC4x3/Properties/s/SpacingOfBoreholes_3zHZ8zixv0VPAwab6YkHPN/DocProperty.xml (hunk 4 above)
Comment (larswik) [reply IFC4x3/Properties/s/SpacingOfBoreholes_3zHZ8zixv0VPAwab6YkHPN/DocProperty.xml]: Done in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

File: IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/s/StaticDischargeHead_0BrAnLHer7kPgv2nOUi7ob/Documentation.md]: Done in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

File: IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md (hunk 6 above)
Comment (larswik) [reply IFC4x3/Properties/s/StaticSuctionLift_1nUwk$OBT5O9pgD19FbFDm/Documentation.md]: Done in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcImprovedGroundTypeEnum/DocEnumeration.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcImprovedGroundTypeEnum/DocEnumeration.xml:14]: Yep!

File: IFC4x3/PropertyEnumerations/d/PEnum_DewateringMethod.xml (hunk 8 above)
Comment (larswik) [reply IFC4x3/PropertyEnumerations/d/PEnum_DewateringMethod.xml]: Changed to USERDEFINED in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

**Comment by larswik** [comment]
> Agree with Sergej's comments.
> 
> > Should this be IfcPositiveLengthMeasure?
> 
> Yes for both accounts.
> 
> > Existing property which is defined as a bounded value (min-max). Is this ok?
> 
> Sounds good to me. As long as it's possible to set the min and the max as the same number.
> 
> > Assuming positive values
> 
> Again, yes. Depth and spacing may not be negative,

Done in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a) and [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

**Comment by larswik** [comment]
> > Check enum descriptions. Ping @JontyDunn and @hessf-ilf
> 
> One possible proposition:
> 
> 1. **SUMPPUMP:**
>    
>    * Utilization of pumps to remove water collected in a sump or low point.
> 2. **WELLPOINT:**
>    
>    * System of closely spaced wells connected to a common header.
> 3. **DEEPWELL:**
>    
>    * Utilization of drilled wells to lower the water table in deeper soil layers.
> 4. **VACUUMWELLPOINTS:**
>    
>    * Wellpoint system augmented by vacuum technology for increased dewatering efficiency.
> 5. **EDUCTORWELLS:**
>    
>    * Usage of eductor pumps to induce flow and lift water for dewatering.
> 6. **OTHERDEWATERINGMETHOD:**
>    
>    * Encompasses custom or specialized dewatering techniques beyond standard categories.

Included in [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a) and [98fe0b2](https://github.com/bSI-InfraRoom/IFC-Specification/pull/678/commits/98fe0b26488006ded4b93c525a4779f670f1433a)

File: IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/Documentation.md (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/n/NumberOfBoreholes_1DBeislK1CbfTPnPGztE29/Documentation.md]: Done!

**Comment by larswik** [comment]
Should be ready for review/merge.

File: IFC4x3/Properties/u/UserDefinedDewateringMethod_1mTiHedsH5Sxf_myHbp4Qm/DocProperty.xml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="UserDefinedDewateringMethod_1mTiHedsH5Sxf_myHbp4Qm" Name="UserDefinedDewateringMethod" UniqueId="7076c468-9f64-4573-ba7e-c3c465cc46b0" PrimaryDataType="IfcLabel" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/u/UserDefinedDewateringMethod_1mTiHedsH5Sxf_myHbp4Qm/DocProperty.xml]: Can we delete this one then?

File: IFC4x3/PropertyConstants/o/OTHERDEWATERINGMETHOD_0XkzIJfrf01A4wQNSilK0U.xml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="OTHERDEWATERINGMETHOD" UniqueId="21bbd493-a75a-4004-a13a-69772cbd401e">
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/o/OTHERDEWATERINGMETHOD_0XkzIJfrf01A4wQNSilK0U.xml]: How about just "OTHER"? Think the pset and property provide enough context.

**Comment by larswik** [comment]
This one seems ok now after changes from @SergejMuhic. I resolved a conflict so that the DEWATERED enum constant is there.

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**